website/utils/url_utils.ts

```diff
@@ -320,66 +386,92 @@ export function deserializeViewerChannelSetting(
   return result;
 }
 
-async function loadDataset(dataset: string, id: string): Promise<Partial<AppProps>> {
-  const db = new FirebaseRequest();
-  const args: Partial<AppProps> = {};
-
-  const datasets = await db.getAvailableDatasets();
-
-  let datasetMeta: DatasetMetaData | undefined = undefined;
-  for (const d of datasets) {
-    const innerDatasets = d.datasets!;
-    const names = Object.keys(innerDatasets);
-    const matchingName = names.find((name) => name === dataset);
-    if (matchingName) {
-      datasetMeta = innerDatasets[matchingName];
-      break;
-    }
-  }
-  if (datasetMeta === undefined) {
-    console.error(`No matching dataset: ${dataset}`);
-    return {};
-  }
-
-  const datasetData = await db.selectDataset(datasetMeta.manifest!);
-  const baseUrl = datasetData.volumeViewerDataRoot + "/";
-  args.imageDownloadHref = datasetData.downloadRoot + "/" + id;
-  // args.fovDownloadHref = datasetData.downloadRoot + "/" + id;
-
-  const fileInfo = await db.getFileInfoByCellId(id);
-  args.imageUrl = baseUrl + fileInfo!.volumeviewerPath;
-  args.parentImageUrl = baseUrl + fileInfo!.fovVolumeviewerPath;
-
-  return args;
+export function serializeViewerChannelSetting(channelSetting: ChannelState): ViewerChannelSettingParams {
+  return removeUndefinedProperties({
+    ven: channelSetting.volumeEnabled ? "1" : "0",
+    sen: channelSetting.isosurfaceEnabled ? "1" : "0",
+    isv: channelSetting.isovalue.toString(),
+    isa: channelSetting.opacity.toString(),
+    clz: channelSetting.colorizeEnabled ? "1" : "0",
+    cza: channelSetting.colorizeAlpha?.toString(),
+    // Convert to hex string
+    col: colorArrayToHex(channelSetting.color),
+    // TODO: Serialize control points....
+    // lut: channelSetting.lut?.join(":"),
+  });
 }
 
-export async function getArgsFromParams(urlSearchParams: URLSearchParams): Promise<{
-  args: Partial<AppProps>;
-  viewerSettings: Partial<ViewerState>;
-}> {
-  const params = urlSearchParamsToParams(urlSearchParams);
-  let args: Partial<AppProps> = {};
-  const viewerSettings: Partial<ViewerState> = {};
+export function deserializeViewerState(params: ViewerStateParams): Partial<ViewerState> {
+  const result: Partial<ViewerState> = {
+    maskAlpha: parseStringInt(params.mask, 0, 100),
```

Comment:
suggestion for readability: create a general `parse` function that takes a parameter, a type and options, that then calls each of these functions. 
`maskAlpha: parse(params.mask, paramType.int, [0, 100])`

or a lookup table from the urlParam enum to the state field with options

Reply:
Tried noodling on this with Cameron... The main blocker was trying to get the same functionality we currently have when parsing enums.  Currently, the `parseStringEnum()` function really nicely guarantees that the returned type is of the enum that was passed in. 

There isn't an easy way with TypeScript to make `parse` generic *and* still support the correct return type for enums passed in as variables.